tests/integration/allocator_accessibility.cpp

```diff
@@ -0,0 +1,175 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "gtest/gtest.h"
+#include "gtest/gtest-death-test.h"
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/config.hpp"
+#include "umpire/util/MemoryResourceTraits.hpp"
+
+//Need to enable gtest death tests with 
+// -DENABLE_GTEST_DEATH_TESTS=On
```

Comment:
This is another thing I'm unsure about. Whenever this test is added to the CMakeLists.txt, this flag must be turned on. Maybe we can add this flag to the Developer Defaults? Or something else to make it automatic (I don't think the tests will run currently)